src/vm/memory/relocatable.zig

```diff
@@ -420,7 +420,7 @@ pub const MaybeRelocatable = union(enum) {
     pub fn tryIntoRelocatable(self: Self) !Relocatable {
```

Comment:
Here, as a result of your refactor, can we enforce a better typing of the output? like:

```zig
pub fn tryIntoRelocatable(self: Self) CairoVMError!Relocatable {
```